File: CS/WinWebSolution.Module/Customer.cs
using System;
using DevExpress.Xpo;
using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;

namespace WinWebSolution.Module {
    [DefaultClassOptions]
    public class Customer : BaseObject {
        public Customer(Session session) : base(session) { }
        private string name;
        public string Name {
            get { return name; }
            set {
                SetPropertyValue("Name", ref name, value);
            }
        }
        private string email;
        public string Email {
            get {
                return email;
            }
            set {
                SetPropertyValue("Email", ref email, value);
            }
        }
        private string phoneNumber;
        public string PhoneNumber {
            get {
                return phoneNumber;
            }
            set {
                SetPropertyValue("PhoneNumber", ref phoneNumber, value);
            }
        }
        protected string CustomerLookupDisplayText {
            get {
                return String.Format("{0} (Email: {1}, PhoneNumber: {2} )", Name, Email, PhoneNumber);
            }
        }
    }
}

File: VB/VB/WinWebSolution.Module/Customer.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo
Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation

Namespace WinWebSolution.Module
	<DefaultClassOptions> _
	Public Class Customer
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private name_Renamed As String
		Public Property Name() As String
			Get
				Return name_Renamed
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", name_Renamed, value)
			End Set
		End Property
		Private email_Renamed As String
		Public Property Email() As String
			Get
				Return email_Renamed
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Email", email_Renamed, value)
			End Set
		End Property
		Private phoneNumber_Renamed As String
		Public Property PhoneNumber() As String
			Get
				Return phoneNumber_Renamed
			End Get
			Set(ByVal value As String)
				SetPropertyValue("PhoneNumber", phoneNumber_Renamed, value)
			End Set
		End Property
		Protected ReadOnly Property CustomerLookupDisplayText() As String
			Get
				Return String.Format("{0} (Email: {1}, PhoneNumber: {2} )", Name, Email, PhoneNumber)
			End Get
		End Property
	End Class
End Namespace